WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift

```diff
@@ -665,6 +665,8 @@ private extension CardReaderConnectionController {
         }
     }
 
+    // TODO: The use of UIViewController here is pretty problematic when using it from SwiftUI,
+    // we need an alternative when it gets consumed through the CardPresentPaymentsAdaptor
```

Comment:
+100, I encountered this when trying to understand all the card reader connection UI states in https://github.com/woocommerce/woocommerce-ios/pull/12723. It also looks like other views have a similar problem with the web view coupled with the business logic and not abstracted out. 😅

Reply:
I can't figure out a way to adapt these without changing the underlying implementation at all right now.

We should be able to ignore most of these webview launching handlers, as it's an implementation detail of the UIKit view model which we can do separately in SwiftUI... but I think we need this one to work pretty soon, because it's essentially an onboarding issue, but handled during connection.

It feels like much of this needs to be moved to the AlertProvider, where we can inject a different one for use from SwiftUI, but the `else` case would need to be handled as that changes the alert shown in the connection controller.